### DISCUSSION TITLE
CSS and JS assets not loaded at all on production (basset issue)

### DISCUSSION THREAD

**Discussion by floristenhove** [Laravel-Backpack/community-forum#855] [Q&A (Help)] [answered]
After deploying my site to a fresh Laravel Forge server the JS and CSS files are not being loaded, they are not even being shown in the actually rendered HTML. Locally everything works fine. 

I've tried a few things already like running `php artisan basset:check`, which gives me three green DONEs. I've even tried running `php artisan backpack:install`, (re)linking the storage folder, running `php artisan basset:fresh`, setting `BASSET_DEV_MODE=true`, checking filesystem disk configuration and a few other things. The `APP_URL` is set correctly and.

I have done some debugging and when I `dd()` the `$href` of the `echoCss` method in BassetManager.php (line 115) I do see the correct path to one of the css files (in this case, `storage/basset/unpkg.com/@digitallyhappy/backstrap@0.5.1/dist/css/legacy.css?f8ca1ce56690`). However, if I just `dump()` instead of dd()`, I don't see anything where locally I do.

Any help is much appreciated. I'm happy to share more details if needed.

---

### PHP VERSION:
PHP 8.3.3-1+ubuntu22.04.1+deb.sury.org+1 (cli) (built: Feb 15 2024 18:38:52) (NTS)
Copyright (c) The PHP Group
Zend Engine v4.3.3, Copyright (c) Zend Technologies
    with Zend OPcache v8.3.3-1+ubuntu22.04.1+deb.sury.org+1, Copyright (c), by Zend Technologies

### LARAVEL VERSION:
10.45.1.0

### BACKPACK PACKAGE VERSIONS:
backpack/basset: 1.2.4
backpack/crud: 6.6.4
backpack/permissionmanager: 7.1.1
backpack/pro: 2.1.7
backpack/settings: 3.1.0
backpack/theme-coreuiv2: 1.2.3

**Comment by karandatwani92** [comment]
Hey @floristenhove

I saw your question on [Stackoverflow](https://stackoverflow.com/questions/78046519/css-and-js-assets-not-loaded-at-all-in-laravel-backpack). Did you check the [FAQ](https://github.com/Laravel-Backpack/basset?tab=readme-ov-file#faq) section?

I tested on a fresh install & worked for me. I still highly doubt on `APP_URL`.
Please ensure that `APP_URL` in your .env matches your server configuration, including the hostname, protocol, and port number.

can you share your current value of `APP_URL`?

**Reply by floristenhove** [reply]
Hi! Sorry for double posting. I figured this might be a better place to post.

I did check the FAQ. I can not share the actual value of the APP_URL, but it's similar to: https://api.domain.com. It matches the server configuration.

**Comment by floristenhove** [comment]
Other details that might be important are that this project is an upgrade from older versions of laravel (5.6) and backpack (3.4). The APP_ENV is set to `uat`.

**Comment by pxpm** [comment]
Hey @floristenhove thanks for the report, and sorry for the trouble. 🙏 

Since it's an upgrade, what I suspect is that you may have files overwritten in your `resources/views` folder that may not bet getting the updated versions ? 

Another thing, what's APP_ENV `uat` ? AFAIK that's to do with node and react, can you try setting the APP_ENV to `production` and use a different .env variable for your javascript ?

I will need some more errors /screenshots/error stacks to help you, so if you can't share the current project I would advise you to setup a `staging` environment in forge too, that you can use to reproduce the issue and share the errors with us. 

Cheers

**Reply by floristenhove** [reply]
Thank you for your help @pxpm. 

I do have some custom views in my `resources/views` dir but I have not overwritten any of the layout files. I have manually updated all other views that I have overwritten.

APP_ENV `uat` is a value I use for our user acceptance test environment. It could be anything really. It is not related to the `NODE_ENV`. Changing it to `production` or `development` does not do anything unfortunately.

The weird thing is that there are no errors at all. Not in the console, not in the logs. It simply does not render any link tags to the css and js. It must be something small and stupid (like always) but I just can't figure it out. 

Any pointers with debugging could also be helpful.

**Comment by floristenhove** [comment, marked as answer]
I've (kinda) resolved it by using a different hostname. My APP_URL was https://api-uat.domain.com, but I've now changed it to https://api.uat.domain.com and everything works. Could the dash be the problem? It's hard to imagine but I'm happy it's working now. Thank you again for your help, it's much appreciated.
